plugin-flex-ts-template-v2/src/feature-library/custom-transfer-directory/custom-components/QueueDirectoryTab.tsx

```diff
@@ -0,0 +1,286 @@
+  const fetchInsightsQueueData = async () => {
+    // check if insights data has been turned off
+    if (!shouldFetchInsightsData()) return;
+
+    // check that the insights client is available
+    if (
+      !ClientManagerInstance.InsightsClient ||
+      ClientManagerHelpers.isForcedDegraded(ClientManagerInstance.InsightsClient)
+    ) {
+      Notifications.showNotification(CustomTransferDirectoryNotification.FailedLoadingInsightsClient);
+      return;
+    }
+
+    // get real time stats map
+    const queueMap = await insightsClient.map({
+      id: 'realtime_statistics_v1',
+      mode: 'open_existing',
+    });
+
+    if (!queueMap) {
+      Notifications.showNotification(CustomTransferDirectoryNotification.FailedLoadingInsightsData);
+
+      return;
+    }
+
+    // make sure all queues are loaded
+    const insightQueues = await getAllSyncMapItems(queueMap);
+
+    // update the queue item
+    queueMap.on('itemUpdated', (updatedItem) => {
+      const {
+        item: { key, data },
+      } = updatedItem;
+
+      const queue = transferQueues.current.find((transferQueue) => transferQueue.sid === key);
+      if (queue && data) {
+        mapRealTimeDataToTransferQueueItem(queue, data as IRealTimeQueueData);
+      }
+
+      filterQueues();
+    });
+
+    // if a queue is added trrigger a reload
+    queueMap.on('itemAdded', () => {
+      fetchSDKTaskQueues();
+    });
+
+    // if a queue is removed trigger a reload
+    queueMap.on('itemRemoved', () => {
+      fetchSDKTaskQueues();
+    });
+
+    setInsightsQueues(insightQueues);
+  };
+
+  // function to resolve fetchedQueues and insights queue data
+  const generateTransferQueueList = () => {
+    const tempQueues = [] as Array<TransferQueue>;
+    fetchedQueues.forEach((value) => {
+      const tempInsightsQueue = insightsQueues.find((item) => item.key === value.sid);
+      const data = tempInsightsQueue?.data as IRealTimeQueueData;
+
+      // merge the fetched queues data with the transfer queue data
+      tempQueues.push(mapRealTimeDataToTransferQueueItem(value as TransferQueue, data));
+    });
+
+    // cache the merged list of fetched queues with real time data
+    transferQueues.current = tempQueues;
+
+    // Apply filter and sort alphabetically
+    filterQueues();
+  };
+
+  // function to filter the generatedQueueList and trigger a rerender
+  const filterQueues = () => {
+    const updatedQueues = transferQueues.current
+      .filter((queue) => {
+        const searchString = searchInputRef.current?.value.toLocaleLowerCase() || '';
+        return queue.name.toLocaleLowerCase().includes(searchString);
+      })
+      .filter((queue) => {
+        if (showOnlyQueuesWithAvailableWorkers()) {
+          // returning only queues with available workers
+          // or queues where meta data is not available
+          return queue.total_available_workers === null || queue.total_available_workers > 0;
```

Comment:
In my account, only the queues that have not seen action in a while have the null value here, so I'm wondering if it makes sense to include those when this setting is used. Maybe if there is an insights client issue, then display the null ones?

Reply:
yeah thats exactly whats happening - if the data is missing we dont know if it should be excluded or not, so just include it (makes it easier to resolve if the issue is an inactive queue, because we can push something into the queue)